File: deep-aerospike-native/src/main/java/com/stratio/deep/examples/scala/WritingCellToAerospikeNative.scala
package com.stratio.deep.examples.scala

import com.stratio.deep.aerospike.direct.config.{AerospikeConfigFactory, AerospikeDeepJobConfig}
import com.stratio.deep.aerospike.direct.extractor.AerospikeNativeCellExtractor
import com.stratio.deep.commons.entity.Cells
import com.stratio.deep.core.context.DeepSparkContext
import com.stratio.deep.examples.utils.ContextProperties
import org.apache.spark.rdd.RDD

/**
 * Simple example for writing a Cell to Aerospike using the native extractor.
 */
object WritingCellToAerospikeNative {

  def main(args: Array[String]) {
    doMain(args)
  }

  def doMain(args: Array[String]) {
    val job: String = "scala:writingCellToAerospikeNative"
    val host: String = "localhost"
    val port: Integer = 3000
    val namespace: String = "test"
    val inputSet: String = "input"
    val outputSet: String = "output"

    val p: ContextProperties = new ContextProperties(args)

    val deepContext: DeepSparkContext = new DeepSparkContext(p.getCluster, job, p.getSparkHome, p.getJars)

    val inputConfigEntity: AerospikeDeepJobConfig[Cells] = AerospikeConfigFactory.createAerospike().host(host).port(port).namespace(namespace).set(inputSet)
    inputConfigEntity.setExtractorImplClass(classOf[AerospikeNativeCellExtractor])
    inputConfigEntity.initialize

    val inputRDDEntity: RDD[Cells] = deepContext.createJavaRDD(inputConfigEntity)

    val outputConfigEntity: AerospikeDeepJobConfig[Cells] = AerospikeConfigFactory.createAerospike().host(host).port(port).namespace(namespace).set(outputSet)
    outputConfigEntity.setExtractorImplClass(classOf[AerospikeNativeCellExtractor])
    outputConfigEntity.initialize

    DeepSparkContext.saveRDD(inputRDDEntity, outputConfigEntity)

    deepContext.stop
  }

}

File: deep-aerospike-native/src/main/java/com/stratio/deep/examples/java/WritingCellToAerospikeNative.java
package com.stratio.deep.examples.java;

import com.stratio.deep.aerospike.direct.config.AerospikeConfigFactory;
import com.stratio.deep.aerospike.direct.config.AerospikeDeepJobConfig;
import com.stratio.deep.aerospike.direct.extractor.AerospikeNativeEntityExtractor;
import com.stratio.deep.core.context.DeepSparkContext;
import com.stratio.deep.examples.utils.ContextProperties;
import org.apache.log4j.Logger;
import org.apache.spark.rdd.RDD;

/**
 * Simple example for writing a Cell to Aerospike using the native extractor.
 */
public class WritingCellToAerospikeNative {

    private static final Logger LOG = Logger.getLogger(WritingCellToAerospikeNative.class);

    private WritingCellToAerospikeNative() {
    }

    public static void main(String[] args) {
        doMain(args);
    }

    public static void doMain(String[] args) {
        String job = "java:writingCellToAerospikeNative";

        String host = "127.0.0.1";
        int port = 3000;

        String namespace = "book";
        String set = "input";

        // Creating the Deep Context where args are Spark Master and Job Name
        ContextProperties p = new ContextProperties(args);
        DeepSparkContext deepContext = new DeepSparkContext(p.getCluster(), job, p.getSparkHome(),
                p.getJars());

        AerospikeDeepJobConfig inputConfigCell = AerospikeConfigFactory.createAerospike().host(host).port(port)
                .namespace(namespace)
                .set(set);
        inputConfigCell.setExtractorImplClass(AerospikeNativeEntityExtractor.class);

        RDD inputRDDCell = deepContext.createRDD(inputConfigCell);

        LOG.info("count : " + inputRDDCell.count());
        LOG.info("prints first cell : " + inputRDDCell.first());

        AerospikeDeepJobConfig outputConfigCell = AerospikeConfigFactory.createAerospike().host(host).port(port)
                .namespace(namespace)
                .set(set).initialize();

        deepContext.saveRDD(inputRDDCell, outputConfigCell);

        RDD outputRDDCell = deepContext.createRDD(outputConfigCell);

        LOG.info("count output : " + outputRDDCell.count());
        LOG.info("prints first output cell: " + outputRDDCell.first());

        deepContext.stop();

    }
}
